scripts/check_docs_formatted.py

```diff
@@ -149,9 +149,20 @@ def format_contents(src: str) -> tuple[str, Sequence[CodeBlockError]]:
     errors: list[CodeBlockError] = []
 
     def _snipped_match(match: Match[str]) -> str:
+        language = match["language"]
+        extension: Literal["py", "pyi"]
+        if language == "python":
+            extension = "py"
+        elif language == "pyi":
+            extension = "pyi"
+        else:
+            # We are only interested in checking the formatting of py or pyi code blocks
+            # so we can return early if the language is not one of these.
+            return f'{match["before"]}{match["code"]}{match["after"]}'
```

Comment:
Could consider using pattern-matching here:

```suggestion
        match language:
            case "python":
                extension = "py"
            case "pyi":
                extension = "pyi"
            case _:
                # We are only interested in checking the formatting of py or pyi code blocks
                # so we can return early if the language is not one of these.
                return f'{match["before"]}{match["code"]}{match["after"]}'
```